src/components/LeaderBoard/TeamLeaderboardChart.tsx

```diff
@@ -153,42 +179,140 @@ const TeamLeaderBoardChart: React.FC<{ data: Array<Data> }> = ({ data }) => {
-    if (box != null) {
-      const newHeight = labelLength * 60;
-      box.setAttribute('style', 'height: ' + newHeight.toString() + 'px');
+  //gets the index to calculate the scoll distance needed to bring the user into view
+  const scrollToUser = () => {
+    const content = contentRef.current;
+    let y = 0;
+    data.every((member: any) => {
+      if (!member.highlight) {
+        y += 1;
+        return true;
+      } else {
+        return false;
+      }
+    });
+    if (content) {
+      content.scrollTop = (y+2) * chartHeightMultiplier;
     }
   };
+  //gets the data from the db for users or teams, sorts them based on highest to lowest steps, and sets the data
+  async function getData(dataType: string) {
+    setLoading(true);
+    const indData: Array<Data> = [];
+    if (dataType == 'teamMembers') {
+      const usersRef = collection(FirestoreDB, 'users'); // get all users reference
+      const q = query(usersRef, where('team', '==', ctx.team)); // query team members
+      const querySnapshot = await getDocs(q); // get team members data
+      querySnapshot.forEach((doc: any) => {
+        const person: Data = {
+          name: doc.data().name as string,
+          profile_pic: doc.data().profile_pic as string,
+          totalStep: doc.data().totalStep as number,
+          highlight:
+            auth.currentUser.email == doc.data().email
+              ? (true as boolean)
+              : (false as boolean)
+        };
+        indData.push(person);
+      });
+      setData(
+        indData.sort((a: any, b: any) => (a.totalStep > b.totalStep ? -1 : 1))
+      );
+    }
+    if (dataType == 'teams') {
+      const querySnapshot = await getDocs(collection(FirestoreDB, 'teams'));
+      querySnapshot.forEach((doc: any) => {
+        const team: Data = {
+          name: doc.data().name as string,
+          profile_pic: doc.data().profile_pic as string,
+          avg_steps: doc.data().avg_steps as number,
+          highlight: false as boolean
+        };
+        const teamMembers = doc.data().members;
+        teamMembers.forEach((member: string) => {
+          if (auth.currentUser.email == member) team.highlight = true;
+        });
+        const today = new Date();
+        const deadline = new Date(adData.teamDate);
+        if (deadline < today) {
+          const membersLength = doc.data().members.length;
+          if (adData.minSize <= membersLength) {
+            indData.push(team);
+          }
+        } else {
+          indData.push(team);
+        }
+        setData(
+          indData.sort((a: any, b: any) => (a.avg_steps > b.avg_steps ? -1 : 1))
+        );
+      });
+    }
+
+    boxAdjust(indData.length);
+    setTimeout(() => {
+      setLoading(false);
+    }, 0);
+  }
+  //do not add data as a redux or you will end up with an infinite loop
+  useEffect(() => {
+    getData(dataType); //go into the firestore and get all the users' names, pictures, and then totalStep
+  }, [dataType]);
```

Comment:
I think the dependency array has to be an empty array here.

Reply:
If this way, is changing the dataType will call getDocs function all the time?

Reply:
I have changed so that there is now a single call in a new function and the data for each chart is maintained instead. Now get data just calls the populated data array.